server/game/cards/19.1-AS01/Dragon/VillageDoshin.js

```diff
@@ -0,0 +1,90 @@
+const DrawCard = require('../../../drawcard.js');
+const {
+    Locations,
+    CardTypes,
+    Players
+} = require('../../../Constants');
+const AbilityDsl = require('../../../abilitydsl');
+
+class VillageDoshin extends DrawCard {
+    setupCardAbilities() {
+        this.wouldInterrupt({
+            title: 'Protect attachment from leaving play',
+            location: Locations.Hand,
+            cost: AbilityDsl.costs.discardSelf(),
+            when: {
+                onCardLeavesPlay: (event, context) => {
+                    let attachment = event.card.type === CardTypes.Attachment;
+                    let youControl = event.card.controller === context.player;
+                    let inPlay = event.card.location === Locations.PlayArea;
+                    let byAbility = event.context.ability.isTriggeredAbility();
+                    let fromOpponent =
+                        event.context.player === context.player.opponent;
+                    return (
+                        attachment &&
+                        youControl &&
+                        inPlay &&
+                        byAbility &&
+                        fromOpponent
+                    );
+                }
+            },
+
+            gameAction: AbilityDsl.actions.conditional({
+                condition: (context) =>
+                    context.player.opponent.fate >= this.doshinTax(context),
```

Comment:
You need to also check if your opponent here can spend fate.
`AbilityDsl.actions.loseFate().canAffect(context.player.opponent, context)`

Reply:
Fixed